### ISSUE TITLE
Allow for loading of AMS whilst printing 

### ISSUE THREAD

**Issue by greghesp** [bambulab/BambuStudio#1804]
**Is your feature request related to a problem? Please describe.**
The scenario is that the AMS is loaded, and a print is started. Mid print you realise you've not got enough filament in that slot and you're about to go out, go to bed etc.

Rather than let the print pause and delay the print, or load a brand new roll into the slot, you want to add the same filament to another slot.

This way, when 1 slot runs out, it'll fail over to the next

**Describe the solution you'd like**
Allow printer to load and unload filament mid print

**Comment by sodachen22** [comment]
@greghesp Can the AMS filament backup feature satisfy the requirement?
Please check the description in the release notes.
https://wiki.bambulab.com/en/x1/manual/X1-X1C-AMS-firmware-release-history
![image](https://github.com/bambulab/BambuStudio/assets/112450380/99796727-f3ad-42c5-8e55-694d260e32c9)

**Comment by KD7LRJ** [comment]
In my experience, if the print was started with a single roll of filament in the AMS, and a second spool is loaded after the print was started (as the OP stated), the AMS will NOT switch to the second roll of filament when the first one is empty even with this option selected.

It also seems that it won't ever switch to a slot to the left of the original slot with this option selected (even if the spools to the left match the original spool), but I'm not sure on this point given recent updates.

**Comment by sodachen22** [comment]
Yes, I can understand the problem, the root cause is AMS does not support read RFID in printing.
We will make the firmware support reading RFID a new filament when it is put to AMS, and support setting the filament when the new filament is not Bambu Lab filament.

**Comment by Hen3V** [comment]
What is the status of this change? I just got my X1C with AMS, and ran into this issue already. I would like to be able to load filaments while printing. I even can't load the AMS when I pause the print.

**Comment by JadenSWang** [comment]
@Hen3V what did you do? Did you have to scrap a print because of this?

**Comment by Hen3V** [comment]
> @Hen3V what did you do? Did you have to scrap a print because of this?

No, but now I had to wait until the spool ran out, and then replace it to be able to continue printing. If I would be able to load the new spool in the AMS already, the AMS would have been able to switch to the other spool automatically during the night. I think this feature would be particularly useful in case you want to add a spool as a backup spool.

**Comment by JadenSWang** [comment]
> > @Hen3V what did you do? Did you have to scrap a print because of this?
> 
> No, but now I had to wait until the spool ran out, and then replace it to be able to continue printing. If I would be able to load the new spool in the AMS already, the AMS would have been able to switch to the other spool automatically during the night. I think this feature would be particularly useful in case you want to add a spool as a backup spool, when you notice the current spool will empty before printing is finished.

Makes sense. I wonder if potentially you could have snipped the filament, waited for it to run out, and replace the dwindling roll with a new one if you needed to overnight.

**Comment by Hen3V** [comment]
@JadenSWang 
I guess that could have worked. But obviously you are then left with a used, almost empty spool, which is less convenient than using all of the spool before switching. But it could be a workaround. Thanks for the hint.

**Comment by thenewguy** [comment]
It would still be helpful as of today if it was possible to add a new roll of filament into the AMS. I just tried and cannot assign the AMS slot a new type of filament while the print is running

**Comment by SirRyanTheGeek** [comment]
Not sure how you got that to work. I pushed the filament into the hole in the AMS 2 Pro, then went into Bambu Studio (closed it and re-opened it again), went to assign that location and got a pop-up with the description “Setting AMS slot information while printing is not supported”.

**Comment by tryagainkid69** [comment]
> Not sure how you got that to work. I pushed the filament into the hole in the AMS 2 Pro, then went into Bambu Studio (closed it and re-opened it again), went to assign that location and got a pop-up with the description “Setting AMS slot information while printing is not supported”.

same with me i have p2s with ams2 pros (2x)

**Comment by Misiu** [comment]
This would be awesome! I'm using an external dryer, so I'd like to start printing with what I have left and when my drying finish I'd like to add extra spool, so the backup function would pick the new spool and continue.

**Comment by JetForMe** [comment]
I too would like to be able to unload and load an unused slot and sync up with BambuStudio to prep for my next print.

**Comment by timeisanillusion** [comment]
+1 
The AMS should at least recheck all slots when a filament runs out, to check if a new spool has been loaded.  
Even if this only works with RFID filaments, it would be really useful and seems like a simple feature.  Even if it only checks "?" spools, this seems possible with the current limitations of not reading during print or allowing edits, just read when a spool empties....
